targets/unity-v2/source/Shared/Internal/Util.cs

```diff
@@ -101,6 +101,23 @@ public static string ReadAllFileText(string filename)
             return _sb.ToString();
         }
 
+        public static T TryEnumParse<T>(string value)
+        {
+            try
+            {
+                return (T) Enum.Parse(typeof(T), value);
+            }
+            catch (InvalidCastException castException)
+            {
+                return default(T);
```

Comment:
There are some fairly significant performance problems with default(T).
It also doesn't always cross compile as expected.
Please add an additional parameter, which is to be used if the parse fails.
That way the caller provides the default.

Reply:
OOps, one more thing!